Backpack/CardButton/Classes/SaveCardButton.swift

```diff
@@ -0,0 +1,254 @@
+            return colors.highlightedIconColorContained
+        case .default:
+            return colors.highlightedIconColorDefault
+        }
+    }
+
+    private func updateBackgroundCircleView() {
+        if viewConfigurator.shouldAddButtonBackground(style: style) {
+            if containedBackgroundCircle.superview == nil {
+                addSubview(containedBackgroundCircle)
+                let circleSize = viewConfigurator.buttonBackgroundSize(size: size)
+                containedBackgroundCircle.layer.cornerRadius = circleSize / 2
+
+                NSLayoutConstraint.activate([
+                    containedBackgroundCircle.widthAnchor.constraint(equalToConstant: circleSize),
+                    containedBackgroundCircle.heightAnchor.constraint(equalToConstant: circleSize),
+                    containedBackgroundCircle.centerXAnchor.constraint(equalTo: centerXAnchor),
+                    containedBackgroundCircle.centerYAnchor.constraint(equalTo: centerYAnchor)
+                ])
+                imageView.map { bringSubviewToFront($0) }
+            }
+        } else {
+            if containedBackgroundCircle.superview != nil {
+                containedBackgroundCircle.removeFromSuperview()
+            }
+        }
+    }
+
+    public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        updateLookAndFeel()
+    }
+}
+
+extension BPKSaveCardButton {
+    public func setChecked(_ newValue: Bool, animated: Bool) {
+        guard animated, let imageView = imageView else {
+            checked = newValue
+            return
+        }
+        if newValue {
+            // Update icon color before animation starts
+            checked = true
+            // 1. resize to 30px
+            // 2. wait
+            // 3. resize back to 24px
+            let transformBefore = imageView.transform
+            UIView.animate(
+                withDuration: BPKDuration.animationDurationSm,
+                delay: 0,
+                options: UIView.AnimationOptions([.beginFromCurrentState, .curveEaseOut])
+            ) {
+                imageView.transform = CGAffineTransformMakeScale(1.25, 1.25)
+            }
+            UIView.animate(
+                withDuration: BPKDuration.animationDurationSm,
+                delay: BPKDuration.animationDurationSm * 2,
+                options: UIView.AnimationOptions([.beginFromCurrentState, .curveEaseIn])
+            ) {
+                imageView.transform = transformBefore
+            }
+        } else {
+            // return to default state with transition
+            UIView.transition(
+                with: self,
+                duration: BPKDuration.animationDurationSm,
+                options: .transitionCrossDissolve
+            ) {
+                self.checked = newValue
+            }
+        }
+    }
+}
+
+private extension BPKSaveCardButton {
```

Comment:
Can we follow the approach we used for BPKBadge / BPKChip? Introducing a colour set of sorts? 
This would allow us to be more flexible and not have a struct grow for each type we add

Reply:
> approach we used for BPKBadge
This is what i found. And there is already a similar function in SaveCardButton

```swift
fileprivate extension BPKBadgeType {
    var textColor: UIColor {
        switch self {
        case .success, .warning, .destructive:
            return BPKColor.textOnLightColor
        case .outline, .strong:
            return BPKColor.textOnDarkColor
        case .normal, .inverse:
            return BPKColor.textPrimaryColor
        case .brand:
            return BPKColor.textPrimaryInverseColor
        }
    }
....
```


## SaveCardButton
```
    private func normalIconColor(style: BPKCardButtonStyle, checked: Bool) -> UIColor {
        let colors = checked ? Self.checkedColor : Self.uncheckedColor

        switch style {
        case .onDark:
            return colors.iconColorOnDark
        case .contained:
            return colors.iconColorContained
        case .default:
            return colors.iconColorDefault
        }
    }
```

The reason to introduce `private struct Colors` is to avoid long switch enum. WIthout it, the enum will be longer. 
Which options do you prefer more? the current one or the two below?
```
// 1.  two switch 
if checked {
   switch style {
   //....
   }
} else {
   switch style {
   //....
   }
}

// 2. longer case 
   switch style {
        case .onDark:
            return checked ? BPKColor.xxx :  BPKColor.xxx 
        case .contained:
            return checked ? BPKColor.xxx :  BPKColor.xxx 
        case .default:
            return checked ? BPKColor.xxx :  BPKColor.xxx 
   }
```

Reply:
update: I will update the struct to follow  BPKChipColors :D

Reply:
Updated in df0786b922594f8af4f92b6162e40afd2f676a75